File: server/server-core-acceptance-tests/src/test/java/uk/co/stevewellington/shoppinglist/actions/AddItemByNameToShoppingListTest.java
package uk.co.stevewellington.shoppinglist.actions;

import com.flextrade.jfixture.annotations.Fixture;
import com.flextrade.jfixture.rules.FixtureRule;
import com.google.inject.Injector;
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;
import uk.co.stevewellington.shoppinglist.infrasctructure.ItemStore;
import uk.co.stevewellington.shoppinglist.model.ItemName;

import static com.google.inject.Guice.createInjector;
import static org.mockito.Mockito.verify;

@RunWith(MockitoJUnitRunner.class)
public class AddItemByNameToShoppingListTest {

    @Rule public FixtureRule fr = FixtureRule.initFixtures();

    @Mock ItemStore itemStore;
    @Fixture ItemName itemName;

    private AddItemByNameToShoppingList addItemByNameToShoppingList;

    @Before
    public void setUp() throws Exception {
        Injector injector = createInjector(binder -> binder.bind(ItemStore.class).toInstance(itemStore));
        addItemByNameToShoppingList = injector.getInstance(AddItemByNameToShoppingList.class);
    }

    @Test public void
    should_create_item_in_store_when_adding_new_item_by_name() {
        addItemByNameToShoppingList.addItemByNameToShoppingList(itemName);
        verify(itemStore).createItemName(itemName);
    }
}

File: server/server-core-acceptance-tests/src/test/scala/uk/co/stevewellington/shoppinglist/actions/AddItemByNameToShoppingListTest.scala
package uk.co.stevewellington.shoppinglist.actions

import com.google.inject.AbstractModule
import com.google.inject.Guice.createInjector
import org.junit.runner.RunWith
import org.mockito.Mockito._
import org.scalatest.junit.JUnitRunner
import org.scalatest.mock.MockitoSugar
import uk.co.stevewellington.shoppinglist.infrasctructure.ItemStore
import uk.co.stevewellington.shoppinglist.model.ItemName
import uk.co.stevewellington.shoppinglist.testutils.{Fixture, Guice, UnitSpec}

@RunWith(classOf[JUnitRunner])
class AddItemByNameToShoppingListTest extends UnitSpec with MockitoSugar with Fixture with Guice {

  val itemStore = mock[ItemStore]
  val itemName = fixture[ItemName]

  val injector = createInjector(new AbstractModule {
    override def configure() = bind(classOf[ItemStore]) toInstance itemStore
  })

  val addItemByNameToShoppingList = get[AddItemByNameToShoppingList] from injector

  describe("When adding an item by name") {
    it("AddItemByNameToShoppingList should create an item in the store") {
      addItemByNameToShoppingList.addItemByNameToShoppingList(itemName)
      verify(itemStore).createItemName(itemName)
    }
  }
}
